Added 2015 Illapel earthquake example

apadaki opened the pull request.

apadaki committed:
modularized python files

apadaki committed:
documented data formatting code, clarified origin_time/rigidity, and cleaned up imports

apadaki committed:
documented code and user parameters, added tide data references, removed unnecessary imports

apadaki committed:
redefined offset parameter

apadaki committed:
documented setplot.py and setrun.py, simplified tide gauge addition, highlighted adjustable hyperparameters

apadaki committed:
created README.md

apadaki committed:
edited moving topography filename

apadaki committed:
edited README.md

apadaki committed:
Add 'illapel2015_chile/' from commit 'b164450aa21f886eaa1f378e45238d8f77e6b8dc'

git-subtree-dir: illapel2015_chile
git-subtree-mainline: e9af92b216b46b5702a7ac2b87ebb888c6aad933
git-subtree-split: b164450aa21f886eaa1f378e45238d8f77e6b8dc

apadaki committed:
added syntax highlighting in README.md

apadaki committed:
added Installation and Usage headings

apadaki committed:
removed .gitignore

mandli commented:
@apadaki This looks good to me.  @rjleveque do you want to take a look at this to make sure this looks good and does not duplicate anything that might be somewhere I do not know about?

rjleveque commented:
@apadaki, thanks for doing this!   It's nice that you download the USGS fault parameters and the gauge observations directly in the `maketopo.py` script. 

As set up, the code takes quite a while to run and doesn't seem to be refining optimally for capturing the signal at the two gauges, e.g. the refinement of the main wave is mostly to the south of gauge 32412.   I think it would be better if you could add some [flagregions](http://www.clawpack.org/flagregions.html) (preferably in the new style) to guide the refinement more, allowing high levels only on the path towards the gauges for this example.  

It might also run faster if you use fewer levels with larger refinement ratios, I notice some levels only refine by 2, you could have fewer levels always refining by 4, for example.  And maybe you don't need such fine levels in this example?  People can add more if they want to, but unless such fine grids are needed to illustrate something, it might be nicer if it runs faster by default.

apadaki commented:
@rjleveque Thank you for your feedback! I'll make sure to implement fewer refinement levels to improve the run time and specify manual regions in a commit sometime today or tomorrow.

apadaki committed:
added flagregions in setrun.py and changed AMR ratios to improve running time

apadaki commented:
@rjleveque I simplified the AMR ratios and added flagregions around each gauge, and the simulation seems to run much faster and give similar results on both gauges. If anything else needs to be changed, please let me know!
